client/boot/index.js

```diff
@@ -161,8 +160,39 @@ function boot() {
 
 	translatorJumpstart.init();
 
-	reduxStore = createReduxStore();
+	localforage.config( {
+		name: 'calypso',
+		storeName: 'calypso_store',
+		description: 'Calypso Storage'
+	} );
+	localforage.getItem( 'redux-state', ( error, initialState ) => {
+		let reduxStore;
+		if ( error ) {
+			debug( 'failed to load initial redux-store state', error );
+			reduxStore = createReduxStore();
+		} else {
+			debug( 'loading redux-store with initial state', initialState );
+			reduxStore = createReduxStore( initialState || {} );
+		}
+
+		reduxStore.subscribe( function() {
+			let state = Object.assign( {}, reduxStore.getState() );
+			//remove dirty parts of the redux tree
+			delete state.sites;
+			delete state.themes;
```

Comment:
This part doesn't look too scalable here. We could mimic the whole reducer tree here with an initialization/pruning tree, which could bring about the same scalable unity of the state itself.

```js
const pruneSites = () => undefined; // could be more intelligent here
const pruneThemes = () => undefined;

const pruneState = state => Object.assign( {}, state, {
	sites: pruneSites( state.sites ),
	themes: pruneThemes( state.themes )
} );
```

or we could make a second iteration here to abstract it further

```js
import { fromPairs } from 'lodash';

const pruners = {
	sites: pruneSites,
	themes: pruneThemes
};

const pruneState = staleState => {
	const prunedState = formPairs( Object
		.keys( pruners )
		.map( key => [ key, pruners[ key ]( state[ key ] ) ] ) );

	return Object.assign( {}, staleState, prunedState );
};
```